src/win32/posix_w32.c

```diff
@@ -356,7 +358,7 @@ int p_open(const char *path, int flags, ...)
 		return  -1;
 	}
 
-	handle = CreateFileW(buf, desired_access, FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, creation_disposition, flags_and_attributes, 0);
+	handle = CreateFileW(buf, desired_access, git_posix_w32__windows_sharemode, NULL, creation_disposition, flags_and_attributes, 0);
```

Comment:
The setting is now applied for all files. This is ok for me, but my initial attempt was to implement this only for files which are read only.